common/src/main/java/com/linecorp/centraldogma/common/PathPatternOptions.java

```diff
@@ -0,0 +1,95 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.centraldogma.internal.Util;
+
+/**
+ * A set of {@link PathPatternOption}s.
+ */
+final class PathPatternOptions {
+
+    private static final Pattern FILE_EXTENSION_PATTERN = Pattern.compile("^.{0,1}[0-9a-z]+$");
+
+    private PathPatternOptions() {}
+
+    /**
+     * Appends "&#47;**" to {@code dirPath}.
+     * Returns the path pattern for matching all file(s) under {@code dirPath}.
+     */
+    public static final Function<String, PathPatternOption> STARTS_WITH = pattern ->
+            new PathPatternOption(pattern,
+                                  dirPath -> {
+                                      checkArgument(Util.isValidDirPath(dirPath), "dir");
+                                      return new DefaultPathPattern(ImmutableSet.of(
+                                              dirPath + (dirPath.endsWith("/") ? "" : "/") + "**"));
+                                  });
+
+    /**
+     * Prepends and appends "&#47;**" to target {@code dirPath}.
+     * Returns the path pattern for matching all file(s) containing {@code dirPath}.
+     */
+    public static final Function<String, PathPatternOption> CONTAINS = pattern ->
+            new PathPatternOption(pattern,
+                                  dirPath -> {
+                                      checkArgument(Util.isValidDirPath(dirPath), "dirPath");
+                                      return dirPath.endsWith("/") ? new DefaultPathPattern(
+                                              ImmutableSet.of("/**" + dirPath + "**"))
+                                                                   : new DefaultPathPattern(
+                                              ImmutableSet.of("/**" + dirPath + "/**"));
+                                  });
+
+    /**
+     * Prepends "&#47;**&#47;" to {@code filename}.
+     * Returns the path pattern for matching file(s) ending in {@code filename}.
+     */
+    public static final Function<String, PathPatternOption> ENDS_WITH = pattern ->
+            new PathPatternOption(pattern,
+                                  filename -> {
+                                      checkArgument(Util.isValidFileName(filename), "filename");
+                                      return new DefaultPathPattern(ImmutableSet.of(filename));
```

Comment:
Should we leave the comment on the line?
```suggestion
                                      // `/**` is added by the constructor of `DefaultPathPattern`
                                      return new DefaultPathPattern(ImmutableSet.of(filename));
```